### DISCUSSION TITLE
Support for ISO 8601 duration format?

### DISCUSSION THREAD

**Discussion by volisoft** [juxt/tick#184] [Q&A] [answered]
Is there support for standard *ISO 8601* duration format?
For example, does `tick` understands strings like  `PT1H2M` and parses it into duration? 
Couldn't find appropriate function for it, and wondering if it's supported or planned for future?

As a side note, the string above is recognized as interval: `(t/interval? "PT1H2M") => true`, which might be an issue since it's not an interval.

**Comment by henryw374** [comment]
java.time supports it to some degree, with Duration/parse and Period/parse. 

the t/interval? thing is weird and comes from the fact that tick used to accept strings as args to just about any function, which it would then try and parse. This has not completely been removed and I don't want to break people further now really. but (t/interval? "foo") is also true because the ITimeSpan protocol is extended to strings, with the intention you could do (t/beginning "...some parseable date string...")

**Comment by volisoft** [comment]
Thanks for the answer. I'm aware about Java support for ISO duration, but was wondering if `tick` supports it. 
I'm asking because one of the advantages of `tick` is that it works both on Java and Javascript platforms. Having uniform interface is very convenient.
If there is no support for the duration format, do you plan to add it?

**Comment by henryw374** [comment, marked as answer]
Ah I see what you mean. Cross platform support of almost all java time is
available via cljc.java-time - definitely these parse fns are there.

No plan to add these to tick atm
